test/publish/test_data_quality.py

```diff
@@ -426,6 +426,36 @@ def test_write_total_source_count(test_session, snapshot):
         assert (rows[3]["total_count"]) == 1000
 
 
+def test_write_sulpub_source_count(test_session, snapshot):
+    # create some random data
+    with test_session.begin() as session:
+        for i in range(0, 1000):
+            session.add(
+                Publication(
+                    doi=f"10.000/00000{i}",
+                    pub_year=2024,
+                    sulpub_json={"a": "b"}
+                    if randint(0, 3) == 1
```

Comment:
Instead of doing it randomly how about every 4th  publication gets wos_json, then our test can be precise.

```python
if i % 4 == 0